### PR TITLE
commit with custom author date

### PR DESCRIPTION
**Pull request by mgarciaisaia**
Honor `git commit --date=...` flag.

Re-submitting the stalled #315

### CODE REVIEW

File: tests/units/test_commit.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+require File.dirname(__FILE__) + '/../test_helper'
+
+class TestCommit < Test::Unit::TestCase
+
+  def setup
+    set_file_paths
+  end
+
+  def test_add
```
Comment (tarcinil) [inline tests/units/test_commit.rb]: This should be in `tests/units/test_lib.rb` to ensure that things in `Git::Lib` are tested where they are expected.

The test name makes more sense as `#test_commit_with_date` in that file.

**Comment by mgarciaisaia** [comment]
Thanks for the feedback, @tarcinil. It seems the project moved really forward since I've submitted the original PR - awesome! 🎉 

I've updated the tests to fit the ones that previously existed. It even seems more clear to me, now.

Let me know what you think 👍

File: tests/units/test_lib.rb [hunk 2]
```diff
@@ -21,7 +21,22 @@ def test_commit_data
...
   end
-  
+
+  def test_commit_with_date
+    create_file("#{@wdir}/test_file_1", 'content tets_file_1')
+    @lib.add('test_file_1')
+
+    author_date = Time.new(2016, 8, 3, 17, 37, 0, "-03:00")
+
+    previous = @lib.commit_data('HEAD')
```
Comment (tarcinil) [inline tests/units/test_lib.rb]: What does this provide?  I do not see it being used anywhere.

**Comment by mgarciaisaia** [comment]
Good catch! It was a leftover from a previous version of the test.

**Comment by mgarciaisaia** [comment]
Sorry - hoped you could rebase it from here. There it is! 👍

**Pull request merged**